coc/static/update_static.py

```diff
@@ -6,229 +6,1482 @@
+
+            name_to_id[(troop_name, village_type)] = _id
+            hold_data = {
+                "_id": _id,
+                "name": self._translate(tid=troop_data.get("TID")),
+                "info": self._translate(tid=troop_data.get("InfoTID")),
+                "TID": {
+                    "name": troop_data.get("TID"),
+                    "info": troop_data.get("InfoTID"),
+                },
+                "production_building": self._translate(tid=production_building),
+                "production_building_level": troop_data.get("BarrackLevel"),
+                "upgrade_resource": self._parse_resource(resource=troop_data.get("UpgradeResource")),
+
+                "is_flying": troop_data.get("IsFlying"),
+                "is_air_targeting": troop_data.get("AirTargets"),
+                "is_ground_targeting": troop_data.get("GroundTargets"),
+
+                "movement_speed": troop_data.get("Speed", 0),
+
+                "attack_speed": troop_data.get("AttackSpeed", 0),
+                "attack_range": troop_data.get("AttackRange", 0),
+                "housing_space": troop_data.get("HousingSpace"),
+                "village": "home" if not village_type else "builderBase",
+            }
+            is_super_troop = troop_data.get("EnabledBySuperLicence", False)
+            is_seasonal_troop = troop_data.get("EnabledByCalendar", False)
+            super_troop_data = None
+            if is_super_troop:
+                super_troop_data = full_super_troop_data.get(troop_name)
+                hold_data["super_troop"] = {"original_id": name_to_id[(super_troop_data["Original"], 0)],
+                                            "original_min_level": super_troop_data["MinOriginalLevel"]}
+            if is_seasonal_troop:
+                hold_data["is_seasonal"] = True
+            hold_data["levels"] = []
+
+            max_townhall_converter = self.lab_to_townhall
+            if troop_data.get("ProductionBuilding") == "Barrack2":
+                max_townhall_converter = self.bb_lab_to_townhall
+
+            for level, level_data in troop_data.items():
+                if not isinstance(level_data, dict):
+                    continue
+                #convert times to seconds, all times for all things will be in seconds
+                upgrade_time_seconds = level_data.get("UpgradeTimeH", 0) * 60 * 60
+
+                if not is_super_troop and not is_seasonal_troop:
+                    required_lab_level = level_data.get("LaboratoryLevel")
+                    required_townhall = max_townhall_converter[level_data.get("LaboratoryLevel")]
+                elif is_super_troop: #for super troops use the original troop's lab level'
+                    original_troop = self.full_troop_data.get(super_troop_data["Original"])
+                    required_lab_level = original_troop.get(level).get("LaboratoryLevel")
+                    required_townhall = max_townhall_converter[required_lab_level]
```

Comment:
do we have the data to add an ability property? so for example super minions range shots are accessible